src/lib/comms/sol-coap.c

```diff
@@ -1295,6 +1299,10 @@ respond_packet(struct sol_coap_server *server, struct sol_coap_packet *req,
         return cb(server, resource, req, cliaddr, (void *)c->data);
     }
 
+    if (server->unknown_handler)
+        return server->unknown_handler(server, req, cliaddr,
+            server->unknown_handler_data);
```

Comment:
hum... just noticed it here: our convention is `data` as the first parameter, not the last. (You can change it now or later and fix all of `sol-coap.h`)